torchgeo/datasets/geonrw.py

```diff
@@ -0,0 +1,334 @@
+
+    * https://ieeexplore.ieee.org/document/9406194
+
+
+    .. versionadded:: 0.6.0
+    """
+
+    splits: ClassVar[list[str]] = ['train', 'test']
+
+    train_list: ClassVar[list[str]] = [
+        'aachen',
+        'bergisch',
+        'bielefeld',
+        'bochum',
+        'bonn',
+        'borken',
+        'bottrop',
+        'coesfeld',
+        'dortmund',
+        'dueren',
+        'duisburg',
+        'ennepetal',
+        'erftstadt',
+        'essen',
+        'euskirchen',
+        'gelsenkirchen',
+        'guetersloh',
+        'hagen',
+        'hamm',
+        'heinsberg',
+        'herford',
+        'hoexter',
+        'kleve',
+        'koeln',
+        'krefeld',
+        'leverkusen',
+        'lippetal',
+        'lippstadt',
+        'lotte',
+        'moenchengladbach',
+        'moers',
+        'muelheim',
+        'muenster',
+        'oberhausen',
+        'paderborn',
+        'recklinghausen',
+        'remscheid',
+        'siegen',
+        'solingen',
+        'wuppertal',
+    ]
+
+    test_list: ClassVar[list[str]] = ['duesseldorf', 'herne', 'neuss']
+
+    classes: ClassVar[list[str]] = [
+        'background',
+        'forest',
+        'water',
+        'agricultural',
+        'residential,commercial,industrial',
+        'grassland,swamp,shrubbery',
+        'railway,trainstation',
+        'highway,squares',
+        'airport,shipyard',
+        'roads',
+        'buildings',
+    ]
+
+    colormap: ClassVar[mcolors.ListedColormap] = mcolors.ListedColormap(
+        [
+            '#000000',  # matplotlib black for background
+            '#2ca02c',  # matplotlib green for forest
+            '#1f77b4',  # matplotlib blue for water
+            '#8c564b',  # matplotlib brown for agricultural
+            '#7f7f7f',  # matplotlib gray residential_commercial_industrial
+            '#bcbd22',  # matplotlib olive for grassland_swamp_shrubbery
+            '#ff7f0e',  # matplotlib orange for railway_trainstation
+            '#9467bd',  # matplotlib purple for highway_squares
+            '#17becf',  # matplotlib cyan for airport_shipyard
+            '#d62728',  # matplotlib red for roads
+            '#e377c2',  # matplotlib pink for buildings
+        ]
+    )
+
+    readers: ClassVar[dict[str, Callable[[str], Image.Image]]] = {
+        'rgb': lambda path: Image.open(path).convert('RGB'),
+        'dem': lambda path: Image.open(path).copy(),
+        'seg': lambda path: Image.open(path).convert('I;16'),
+    }
+
+    modality_filenames: ClassVar[dict[str, Callable[[list[str]], str]]] = {
+        'rgb': lambda utm_coords: '{}_{}_rgb.jp2'.format(*utm_coords),
+        'dem': lambda utm_coords: '{}_{}_dem.tif'.format(*utm_coords),
+        'seg': lambda utm_coords: '{}_{}_seg.tif'.format(*utm_coords),
+    }
+
+    modalities: ClassVar[list[str]] = ['rgb', 'dem', 'seg']
+
+    url: ClassVar[str] = (
+        'https://huggingface.co/datasets/torchgeo/geonrw/resolve/main/{}'
+    )
```

Comment:
```suggestion
    url = 'https://hf.co/datasets/torchgeo/geonrw/resolve/3cb6bdf2a615b9e526c7dcff85fd1f20728081b7/{}'
```
Stable git commit hash for reproducibility